x-pack/plugins/cases/server/saved_object_types/migrations/comments.ts

```diff
@@ -119,39 +119,54 @@ export const createCommentsMigrations = (
   return mergeSavedObjectMigrationMaps(commentsMigrations, embeddableMigrations);
 };
 
-export const migrateByValueLensVisualizations =
-  (migrate: MigrateFunction): SavedObjectMigrationFn<{ comment?: string }, { comment?: string }> =>
-  (doc: SavedObjectUnsanitizedDoc<{ comment?: string }>, context: SavedObjectMigrationContext) => {
-    if (doc.attributes.comment == null) {
-      return doc;
-    }
+export const migrateByValueLensVisualizations = (
+  migrate: MigrateFunction,
+  migrationVersion: string
+): SavedObjectMigrationParams<{ comment?: string }, { comment?: string }> => {
+  const deferred = isDeferredMigration(MIN_DEFERRED_KIBANA_VERSION, migrationVersion);
 
-    try {
-      const parsedComment = parseCommentString(doc.attributes.comment);
-      const migratedComment = parsedComment.children.map((comment) => {
-        if (isLensMarkdownNode(comment)) {
-          // casting here because ts complains that comment isn't serializable because LensMarkdownNode
-          // extends Node which has fields that conflict with SerializableRecord even though it is serializable
-          return migrate(comment as SerializableRecord) as LensMarkdownNode;
-        }
-
-        return comment;
-      });
-
-      const migratedMarkdown = { ...parsedComment, children: migratedComment };
+  return {
+    // @ts-expect-error: remove when core changes the types
```

Comment:
@rudolf Is it ok to set it as true when needed?

Reply:
yes, until https://github.com/elastic/kibana/issues/152807 is merged there's a caveat: if there's partial updates against this saved object it could cause migrations to run twice (update `comment` without updating `typeVersion`) or never run (updating `typeVersion` without updating `comment`).

Looking at https://github.com/elastic/kibana/blob/main/x-pack/plugins/cases/common/api/cases/comment/index.ts#L250C5-L261 it doesn't seem like we allow partial updates?

Otherwise we're blocked on https://github.com/elastic/kibana/issues/152807

Reply:
Yes, this is correct. We do not allow partial updates for comments.